### ISSUE TITLE
Video players with controls that auto-dismiss

### ISSUE THREAD

**Issue by scottaohara** [w3c/wcag#2007]
I would like some clarity on what the expectation is for 2.4.7 Focus Visible in regards to video players where the controls, which may have focus, are visually hidden after a timeout period.

The normative text alone seems to indicate that so long as the focusable controls have a focus indicator, then they should be a-ok.

The Understanding Document's Intent section contains the following:
> The focus indicator must not be time limited, when the keyboard focus is shown it must remain.

As that text could be interpreted to mean that if an element is focused, then it must not be allowed to disappear from view.  While I realize the text specifically says "the focus indicator must not be time limited..." if the entire control were to disappear, then so to would the focus indicator.   This interpretation is troubling in regards to video player controls where the expectation would be that controls do get out of the way so people can have an unobstructed view of the video stream, and and would not expect a focus indicator to remain around invisible controls, keep the controls permanently in view, or even be shown around the video stream, for instance, as that could also then be distracting to people watching the video.

FWIW, I can also see an interpretation of the text  (as it was mine until earlier today when I was discussing this with someone) to mean that the focus indicator on the specific control must not be the thing that is time limited, and that so long as keyboard focus remains on that control, and the focus indicator were to remain on the control when / if the controls were all re-revealed by the user, then this would meet the intent of this SC.

I would appreciate the working group's thoughts on this, thanks.

**Comment by patrickhlauke** [comment]
> FWIW, I can also see an interpretation of the text (as it was mine until earlier today when I was discussing this with someone) to mean that the focus indicator on the specific control must not be the thing that is time limited, and that so long as keyboard focus remains on that control, and the focus indicator were to remain on the control when / if the controls were all re-revealed by the user, then this would meet the intent of this SC.

this is my interpretation as well. the SC does not seem to suggest that once something has focus, both the control and the focus indicator must persist, so if the control itself disappears after a while, so can the focus indicator.

**Comment by patrickhlauke** [comment]
related, would be interested to see how this scenario (video player controls that appear when needed, but disappear again after a period of inactivity) would play out against the proposed WCAG 2.3 SC 3.2.7 Visibile Controls https://www.w3.org/TR/WCAG22/#visible-controls

**Comment by mraccess77** [comment]
@patrickhlauke The video controls aspect for Visible controls is already discussed in the understanding doc .

> The presence of the complex component — for example, the carousel component or the video image — provides the “Information needed to identify the user interface components.” For a carousel, controls displayed on hover should also be displayed when the carousel is activated. For a video player, activating the video image should launch the video (sometimes in a new window), at which point more video player controls, including controls displayed previously on hover, should be visible. Some users may still struggle if video controls are not persistently visible, so there is benefit to providing a mechanism that allows users to have the controls persistently visible. 

https://www.w3.org/WAI/WCAG22/Understanding/visible-controls.html

I agree on the 2.4.7 aspect - if the control disappears focus disappears too!

**Comment by patrickhlauke** [comment]
> The video controls aspect for Visible controls is already discussed in the understanding doc

ah, good point. i assume what the understanding there says about "hover" would also apply the same way to "focus" (though having the control and its indicator disappear after a while may be even more disruptive to users than just having the control disappear while the mouse pointer that made it appear on hover still remains there)

**Comment by scottaohara** [comment]
Thanks for the quick back and forth on this, Patrick and Jonathan.  

Sample size of one obviously does not mean there's an actual issue here then, though I wonder if everyone would agree that it's unreasonable for someone to come to the differing interpretation I reported on?  

If people think this is clear enough, then I'm happy for this issue to be closed.  If specifically calling out that if the control were to disappear, then of course the focus indicator would disappear as well, and it would not be expected for focus to otherwise be visible in such a case, then I would be happy with that call out being added too.

**Comment by patrickhlauke** [comment]
"leave no one behind" ... i'll add this to my pile of "things i might submit a PR for that clarify things in understanding docs"

**Comment by patrickhlauke** [comment]
i *think* this is now covered to an extent in "Embedded controls" https://raw.githack.com/w3c/wcag/bc3b52b54f67c6ae57d7defae99c08a4a02985ac/understanding/22/visible-controls.html
